### ISSUE TITLE
Fatal error in backend after update to LLVM 9.0

### ISSUE THREAD

**Issue by rgal** [SNSystems/llvm-project-prepo#81]
Found building LLVM (instructions from https://github.com/SNSystems/llvm-project-prepo/wiki/Build-LLVM-with-the-Repo-Compiler) after merging LLVM 9.0.

The crash happens on the second compile. Note that this is hit on the first time building LLVM with the repo compiler. I assume that the program repo is finding some repeated code between files but I was able to reproduce it building the same file twice.

Testcase reduced from pstore/lib/os/memory_mapper.cpp.
```
$ cat test.cpp
class base {
public:
  ~base();
};
class derived : base {
  ~derived();
};
base::~base() = default;
derived::~derived() = default;
$ rm clang.db
# Build 1 okay
$ clang++ --target=x86_64-pc-linux-gnu-repo -c test.cpp -O1
# Build 2 !okay
$ clang++ --target=x86_64-pc-linux-gnu-repo -c test.cpp -O1
Alias must point to a definition
void (%class.derived*)* @_ZN7derivedD1Ev
fatal error: error in backend: Broken module found, compilation aborted!
clang-9: error: clang frontend command failed with exit code 70 (use -v to see invocation)
clang version 9.0.0 (https://github.com/SNSystems/llvm-project-prepo.git 096cdb0d99c217e00c9a763edb4bf631364a47db)
Target: x86_64-pc-linux-gnu-repo

```

**Comment by MaggieYingYi** [comment]
Triage:

Generate the IR code for the Build 2.
```
$ rm clang.db
# Build 1
$ clang++ --target=x86_64-pc-linux-gnu-repo -c test.cpp -O1
# Build 2
$ clang++ --target=x86_64-pc-linux-gnu-repo -S -emit-llvm test.cpp -O1 -o second.ll
$ cat second.ll
; ModuleID = 'test.cpp'
source_filename = "test.cpp"
target datalayout = "e-m:r-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-pc-linux-gnu-repo"

%class.derived = type { i8 }

@_ZN7derivedD1Ev = unnamed_addr alias void (%class.derived*), void (%class.derived*)* @_ZN7derivedD2Ev

declare void @_ZN7derivedD2Ev(%class.derived*)

!llvm.module.flags = !{!0}
!llvm.ident = !{!1}
!repo.tickets = !{!2, !3, !4, !5}

!0 = !{i32 1, !"wchar_size", i32 4}
!1 = !{!"clang version 9.0.0 (https://github.com/SNSystems/llvm-project-prepo.git 096cdb0d99c217e00c9a763edb4bf631364a47db)"}
!2 = !TicketNode(name: "_ZN4baseD2Ev", digest: [16 x i8] c"4\12hVP\AA\BE\BD)\D1\03\C5\16\E8.`", linkage: external, visibility: default, pruned: true)
!3 = !TicketNode(name: "_ZN4baseD1Ev", digest: [16 x i8] c"4\12hVP\AA\BE\BD)\D1\03\C5\16\E8.`", linkage: external, visibility: default, pruned: true)
!4 = !TicketNode(name: "_ZN7derivedD2Ev", digest: [16 x i8] c"4\12hVP\AA\BE\BD)\D1\03\C5\16\E8.`", linkage: external, visibility: default, pruned: true)
!5 = !TicketNode(name: "_ZN7derivedD1Ev", digest: [16 x i8] c"4\12hVP\AA\BE\BD)\D1\03\C5\16\E8.`", linkage: external, visibility: default, pruned: true)
```
The '_ZN7derivedD1Ev' references the pruned '_ZN7derivedD2Ev'. It should be changed to the declaration. However, it doesn't. This caused to the backend issue.

To solve the issue, the aliases' pruning is separated into two steps. First, check whether a GlobalAlias references a pruned GO. If yes, record it. Second, replace all recorded aliases by the declaration.

**Comment by MaggieYingYi** [comment]
The patch is under the code review PR #82.

**Comment by MaggieYingYi** [comment]
Fixed in the git commit: 570532b.
